client/utils.go

```diff
@@ -31,10 +31,17 @@ func decodeBody(obj interface{}, body io.Reader) error {
 	return nil
 }
 
-func ensureCloseReader(resp *Response) {
+func ensureCloseReader(resp *Response) error {
 	if resp != nil && resp.Body != nil {
 		// close body ReadCloser to make Transport reuse the connection
-		io.CopyN(ioutil.Discard, resp.Body, 512)
-		resp.Body.Close()
+
+		if _, err := io.CopyN(ioutil.Discard, resp.Body, 512); err != nil {
+			return fmt.Errorf("failed to ensure close reader: %v", err)
```

Comment:
IMOP, `return err` is enough. In addition, the caller should also handle the error that this function return.